rust/connlib/clients/apple/src/lib.rs

```diff
@@ -174,10 +176,14 @@ fn init_logging(log_dir: PathBuf, log_filter: String) -> file_logger::Handle {
 }
 
 impl WrappedSession {
+    // TODO: Some of these arguments could be combined into a ConnectInfo struct or something
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Probably needs some refactoring soon

Reply:
I wouldn't worry about it too much for now. The real problem here is that we are passing too much information "down" into the tunnel only to pass it back "up" as part of making an HTTP call to the portal.

Similarly to the gateway, I think the solution here is to separate the components better. Decouple communication with the portal from the tunnel and session. This will result in some code "duplication" between the clients but I think that is better than passing several optional parameters into the session. The fact that we want to customize individual clients deep down in the stack IMO is a sign that we are trying too hard to deduplicate things here.

Reply:
> Decouple communication with the portal from the tunnel and session

I think I see where you're coming from, but I'm not sure I'm quite following what you mean by this. I was thinking the Session is the tunnel + portal coms. We could make these parameters a callback instead, and the relevant platforms could call them to get the info they need as opposed to passing `None` for those platforms.

That might be the better approach here. Thoughts @conectado ?

Reply:
I think what thomas is suggesting is: similarly to the gateway, we pass to the `Session` a `PhoenixChannel` that we already created(+ the init parameters) which I'd be up for, we can do it as part of refactoring the `control` module out of `client-shared`. This makes sense so that `connect` would only need to take that channel and the `CallbackHandler` (we can also just setup the logging before calling `connect` as well)

But we can do that later, for now I think it'd be simpler in a next PR if we could just pass the `user_agent` as part of `connect` so that we don't need to juggle so many parameters.

Reply:
@conectado Yeah passing `user_agent` in connect is a good call, that'd allow us to use the app version number instead of connlib's.